modules/@apostrophecms/util/ui/src/util.js

```diff
@@ -128,7 +128,20 @@ export default () => {
   // THAT ONE WIDGET and NO OTHER. Don't worry about finding the
   // others, we will do that for you and we guarantee only one call per widget.
 
-  apos.util.widgetPlayers = {};
+  const widgetPlayersConfig = {
+    list: {},
+    initialized: false
+  };
+  apos.util.widgetPlayers = new Proxy(widgetPlayersConfig.list, {
+    set(target, prop, value) {
+      target[prop] = value;
+      // run the player if we missed the initial run
+      if (widgetPlayersConfig.initialized) {
+        apos.util.runPlayers(document, { [prop]: value });
```

Comment:
runPlayers doesn't take a second argument. It sounds like you're suggesting one unofficially but I'd rather we not have this bite us later if we add one without looking at this. The lack of specificity won't hurt because elements already played won't get played again.

Reply:
This is a guard against attempting all players again. The internal player guard is not working well in many cases. We also need to look at the refresh calls in the Context Vue component someday, it causes some of the issues I observe @boutell

Reply:
See `apos.util.runPlayers = function (el, newPlayer) {`. We end up with players being called back to back and breaking a lot of stuff. Again, now we have a lot more room for racing conditions and it bites us (mostly in edit mode, when we have also ajax dom updates from what it seems and manipulate the dom while players are attempting to run).